### ISSUE TITLE
--stdin is too picky about filenames

### ISSUE THREAD

**Issue by gurgeous** [bbatsov/rubocop#5789]
I'm working on the vscode-ruby extension, which happily uses rubocop for linting (thanks guys!). If the filename passed to `--stdin` doesn't have a ruby extension (like `.rb`) rubocop won't inspect the file.

I would think that by using --stdin I'm claiming that stdin is filled with lovely ruby code ready for inspection. Regardless of the filename passed to --stdin. The current behavior can be undesirable, since new/untitled files won't get linted. Custom ruby file types also won't get linted.

I could work around this problem by always passing `--stdin file.rb`, or perhaps by appending `.rb` if missing. But that seems to defeat the purpose a bit.

--------

## Expected behavior

`echo def | rubocop --stdin Untitled` should inspect the files and product errors, like:

```sh
$ echo def | rubocop --stdin Untitled
Inspecting 1 file
E
...
Offenses:
Untitled:2:1: E: Lint/Syntax: unexpected token $end
...
```

## Actual behavior

```sh
$ echo def | rubocop --stdin Untitled
Inspecting 0 files
0 files inspected, no offenses detected
```

## Steps to reproduce the problem

`echo def | rubocop --stdin Untitled`

## RuboCop version

```sh
$ rubocop -V
0.54.0 (using Parser 2.5.1.0, running on ruby 2.5.0 x86_64-darwin16)
```

**Comment by Darhazer** [comment]
It's kind of a duplicate of #4666 (the underlying issue is the same)

**Comment by bbatsov** [comment]
Yeah, it's the same issue and we should certainly fix it.
